HLTrigger/Muon/src/HLTMuonDimuonL3Filter.cc

```diff
@@ -199,55 +206,63 @@ HLTMuonDimuonL3Filter::hltFilter(edm::Event& iEvent, const edm::EventSetup& iSet
      edm::Handle<reco::MuonTrackLinksCollection> links;
      iEvent.getByToken(linkToken_, links);
 
+     edm::Handle<trigger::TriggerFilterObjectWithRefs> level1Cands;
+     std::vector<l1t::MuonRef> vl1cands;
+     std::vector<l1t::MuonRef>::iterator vl1cands_begin;
+     std::vector<l1t::MuonRef>::iterator vl1cands_end;
+
+     bool check_l1match = true;
+
      // Loop over RecoChargedCandidates:
      for(unsigned int i(0); i < mucands->size(); ++i){
 	RecoChargedCandidateRef cand(mucands,i);
+        TrackRef tk = cand->track(); // is inner track
+	check_l1match = true;
+        int nlink = 0;
 	for(auto const & link : *links){
-	  TrackRef tk = cand->track();
+	  nlink++;
 
 	  // Using the same method that was used to create the links between L3 and L2
 	  // ToDo: there should be a better way than dR,dPt matching
-	  const reco::Track& globalTrack = *link.globalTrack();
-	  float dR2 = deltaR2(tk->eta(),tk->phi(),globalTrack.eta(),globalTrack.phi());
-	  float dPt = std::abs(tk->pt() - globalTrack.pt())/tk->pt();
-          const TrackRef staTrack = link.standAloneTrack();
-	  if (dR2 < 0.02*0.02 and dPt < 0.001 and previousCandIsL2_) {
+	  const reco::Track& trackerTrack = *link.trackerTrack();
+
+	  float dR2 = deltaR2(tk->eta(),tk->phi(),trackerTrack.eta(),trackerTrack.phi());
+	  float dPt = std::abs(tk->pt() - trackerTrack.pt())/tk->pt();
```

Comment:
careful about pt=0 tracks here. (was already an issue in the old code I realize.)